### DISCUSSION TITLE
Serveur Cloudlink

### DISCUSSION THREAD

**Discussion by Eaielectronic** [community/community#123937] [Programming Help]
### Select Topic Area

Question

### Body

Good morning,
I would like to know how to make Turbowrap's cloudlink function work on my personal server (docker). The server is in python.

**Comment by RysanekDavid** [comment]
Hey there,

**First create a Dockerfile** 
Some example:


```
# Use the official Python image from the Docker Hub
FROM python:3.9-slim

# Set the working directory
WORKDIR /app

# Copy the requirements file into the container
COPY requirements.txt .

# Install any dependencies
RUN pip install --no-cache-dir -r requirements.txt

# Copy the rest of the application code
COPY . .

# Expose the port the app runs on
EXPOSE 8000

# Run the application
CMD ["python", "app.py"]
```

**Create a requirements.txt file - example:**

```
turbowrap
flask
requests

```

**Implement the Cloudlink Function**

**Example:**

```
from flask import Flask, request, jsonify
import turbowrap

app = Flask(__name__)

@app.route('/cloudlink', methods=['POST'])
def cloudlink():
    try:
        data = request.json
        # Assuming you have a cloudlink function in turbowrap
        response = turbowrap.cloudlink(data)
        return jsonify(response)
    except Exception as e:
        return jsonify({'error': str(e)}), 500

if __name__ == '__main__':
    app.run(host='0.0.0.0', port=8000)

```

Build:  
**docker build -t my-python-app .**

Then run: 
**docker run -p 8000:8000 my-python-app**


Hope it will help, David

**Comment by LiteBrite82** [comment]
Thanks for posting in the GitHub Community, @Eaielectronic ! 

We’ve moved your post to our Programming Help 🧑‍💻 category, which is more appropriate for this type of discussion.

Please review our [guidelines](https://github.com/orgs/community/discussions/51384) about the Programming Help category for more information.

**Comment by Eaielectronic** [comment]
Thank
